docroot/modules/custom/va_gov_vet_center/src/EventSubscriber/EntityEventSubscriber.php

```diff
@@ -73,11 +76,146 @@ public function buildNearbyVcAndOutStationFieldsetContent(FormIdAlterEvent $even
+    $form = &$event->getForm();
+    $form_state = $event->getFormState();
+    $form_id = $event->getFormId();
+
+    if ($form_id === 'node_vet_center_cap_form' || $form_id === 'node_vet_center_cap_edit_form') {
+      // Add after_build callbacks for VC CAP node forms.
+      $form['field_address']['widget']['#after_build'][] = 'va_gov_vet_center_vc_cap_address_alter_label_after_build';
+      $form['field_facility_hours']['widget']['#after_build'][] = 'va_gov_vet_center_vc_cap_hours_hide_caption_after_build';
+    }
+
+    if ($form_id === 'node_vet_center_locations_list_form' || $form_id === 'node_vet_center_locations_list_edit_form') {
+      $this->addFacilitiesListingBlockToForm($form);
+    }
+
+    $this->optInCapHours($form, $form_state, $form_id);
+
+    // List of forms to modify media library widget help text.
+    $media_widget_content_form_ids = [
+      'node_vet_center_cap_form',
+      'node_vet_center_cap_edit_form',
+      'node_vet_center_edit_form',
+      'node_vet_center_form',
+      'node_vet_center_mobile_vet_center_form',
+      'node_vet_center_mobile_vet_center_edit_form',
+      'node_vet_center_outstation_form',
+      'node_vet_center_outstation_edit_form',
+    ];
+
+    // We want to modify media library widget help text.
+    if (in_array($form_id, $media_widget_content_form_ids)) {
+      $form['field_media']['widget']['#field_prefix']['empty_selection'] = [
+        '#markup' => t('Add a photo of the facility'),
+      ];
+    }
+    // Require message on revision.
+    $this->requireRevisionMessage($form, $form_state, $form_id);
+  }
+
+  /**
+   * Output facility listing view on vc locations node forms.
+   *
+   * @param array $form
+   *   The form array.
+   */
+  public function addFacilitiesListingBlockToForm(array &$form) {
+    $form['group_my_locations'] = [
+      '#type' => 'details',
+      '#title' => t('Main and satellite locations'),
+      '#open' => TRUE,
+      '#weight' => 3,
+    ];
+    $form['group_my_locations']['vc_facility_listing_view'] = [
+      '#type' => 'view',
+      '#name' => 'vet_center_facility_listing',
+      '#display_id' => 'vc_listing_for_node_form',
+      '#embed' => TRUE,
+    ];
+  }
+
+  /**
+   * Determine whether or not user can edit community access point office hours.
+   *
+   * @param array $form
+   *   The form array.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   The form state.
+   * @param string $form_id
+   *   The form id.
+   */
+  public function optInCapHours(array &$form, FormStateInterface $form_state, $form_id) {
+    if (($form_id === 'node_vet_center_cap_edit_form') || ($form_id === 'node_vet_center_cap_form')) {
+      // We want to implement the states logic only if the field_office_hours
+      // and the field_vetcenter_cap_hours_opt_in are available.
+      /** @var \Drupal\node\NodeInterface $node*/
+      $node = $form_state->getFormObject()->getEntity();
```

Comment:
```suggestion
        /** @var \Drupal\Core\Entity\EntityFormInterface $form_object */
    $form_object = $form_state->getFormObject();
    /** @var \Drupal\node\NodeInterface $node*/
    $node = $form_object->getEntity();
```
phpstan is not so good with chains of calls that each return a different type of object.  Unfortuanately you have to do it in two jumps.

Reply:
Thank you @swirtSJW ! Updated...